### DISCUSSION TITLE
Cleanly using conan packages in meson

### DISCUSSION THREAD

**Discussion by wbehrens-on-gh** [mesonbuild/meson#9944] [Q&A]
I'm trying to setup a system for cleanly using conan packages in meson(I would use wrapDB but the packages I need aren't in there yet) I added the conan install as a run_command and now I need a way to use the install .pc files in meson. I installed them to meson.build_root() and I though just changing the PKG_CONFIG_PATH to that would work but it seems it didn't. Does anyone have and insight on the cleanest way to do this?

![image](https://user-images.githubusercontent.com/35979547/152464418-43904a18-28c4-4c3d-a01f-4c75c949666a.png)
![image](https://user-images.githubusercontent.com/35979547/152464439-f1eed7f4-5e11-4077-aa29-e56b4271e729.png)

**Comment by eli-schwartz** [comment]
> I though just changing the PKG_CONFIG_PATH to that would work but it seems it didn't.

It would, but you didn't change it.

You created a meson variable wrapping a dict which contains an element named PKG_CONFIG_PATH and which is suitable for passing as the `env` kwarg to various functions, such as `custom_target` or `test`.

That's not capable of modifying the original shell environment you ran meson under.

**Comment by wbehrens-on-gh** [comment]
is there a way I can tell meson/pkg-config to look in that directory while keeping it all contained inside meson?

**Reply by wbehrens-on-gh** [reply]
Using the python module in meson you can in theory set environment variables via that but they dont effect the meson task and thus dont work
